pkg/registry/core/componentstatus/rest.go

```diff
@@ -77,13 +77,21 @@ func (rs *REST) NewList() runtime.Object {
 // Returns the list of component status. Note that the label and field are both ignored.
 // Note that this call doesn't support labels or selectors.
 func (rs *REST) List(ctx context.Context, options *metainternalversion.ListOptions) (runtime.Object, error) {
-	servers := rs.GetServersToValidate()
+	servers, err := rs.GetServersToValidate()
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		for _, s := range servers {
+			s.Close()
```

Comment:
do we want to close and recreate each time?

Reply:
Yes, this is how it previously work, we will not change it.